### PR TITLE
Foorm validation follow up (blocks saving for lint errors)

### PR DESCRIPTION
**Pull request by tim-dot-org**
- Adds a parallel hasLintError prop (because the linting happens asynchronously through different callbacks, reusing hasJSONError for the same thing doesn't work correctly)
- Intentionally not using hasLintError for some messages (i.e. preview error message) because we can still render with a lint error.
- Should properly prevent saving.
- Works in survey and library mode.

## Links

- original pr: [40557](https://github.com/code-dot-org/code-dot-org/pull/40557)
- jira ticket: [PLC-1202](https://codedotorg.atlassian.net/browse/PLC-1202)

## Testing story

Tested locally:
verified adding spaces / brackets to names prevented displayed error and prevented saving

### CODE REVIEW

File: apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx [hunk 1]
```diff
@@ -260,24 +261,27 @@ class FoormFormSaveBar extends Component {
         <div style={styles.saveButtonBackground} className="saveBar">
           {this.props.lastSaved &&
             !this.props.saveError &&
-            !this.props.hasJSONError && (
+            !this.props.hasJSONError &&
+            !this.props.hasLintError && (
```
Comment (bencodeorg) [inline apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx]: It looks like `hasLintError` has to be true everywhere that `hasJSONError` has to be true (and the reverse) -- could we add a little helper function that combines them and use that throughout (eg, hasCodeMirrorError()?)

**Review by bencodeorg — APPROVED** [review]
Nice!

File: apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx (hunk 1 above)
Comment (tim-dot-org) [reply apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx]: There's a couple places I wanted to specifically not combine them (i.e. the preview can still render if there is a lint error but no json error) but yeah, i could add a helper for the rest of the cases

Comment (bencodeorg) [reply apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx]: In this component (and the library save bar one) it looks like they're always used in combination, right? I feel like worth having a helper in those?

**Comment by bencodeorg** [comment]
Oh one general comment -- could you link to the original PR that this builds on top of in the PR description? And maybe tidy up the description generally?

**Review by bencodeorg — APPROVED** [review]

**Pull request merged**